streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -972,8 +970,8 @@ public synchronized State state() {
         return state;
     }
 
-    private synchronized void setStateWhenNotInPendingShutdown(final State newState) {
-        if (state == State.PENDING_SHUTDOWN) {
+    private synchronized void setStateWhenNotShuttingDownOrDead(final State newState) {
+        if (state == State.PENDING_SHUTDOWN || state == State.DEAD) {
```

Comment:
Could you describe how https://issues.apache.org/jira/browse/KAFKA-5372 could happen on the first place? Since we only set the state to `DEAD` after the end of `shutdown` call, at which the consumer has already been closed so all?

Also in line 601, should we really call `setState(State.PENDING_SHUTDOWN);` there? Or should we only rely on `close()` call being triggered to set the state?